src/ec/packages/ec-component-card/card.story.js

```diff
@@ -2,62 +2,205 @@
 import { storiesOf } from '@storybook/html';
 import { withNotes } from '@ecl-twig/storybook-addon-notes';
 import withCode from '@ecl-twig/storybook-addon-code';
+import { withKnobs, text, array } from '@storybook/addon-knobs';
+import merge from 'deepmerge';
 
 import defaultSprite from '@ecl/ec-resources-icons/dist/sprites/icons.svg';
 import dataCard from './demo/data--card';
-import dataCardEvent from './demo/data--card-event';
-import dataCardTag from './demo/data--card-tag';
 import dataTile from './demo/data--tile';
+import dataCardTag from './demo/data--card-tag';
+import dataCardEvent from './demo/data--card-event';
 
 import card from './ecl-card.html.twig';
 import notes from './README.md';
 
-const formatInfo = data => {
-  if (data.card.infos) {
-    data.card.infos.forEach(item => {
-      item.icon.path = defaultSprite;
-    });
-  }
+function prepareMerge(oldArrayName, newArrayName, label) {
+  oldArrayName.forEach((a, key) => {
+    if (label === 'label') {
+      newArrayName.push(oldArrayName[key].label);
+    } else {
+      newArrayName.push(oldArrayName[key]);
+    }
+  });
+  oldArrayName.splice(0, oldArrayName.length);
+}
+
+const dataCardInfos = [];
+const dataCardMeta = [];
+const dataCardTags = [];
+const dataTileLinks = [];
+const dataCardTagMeta = [];
+const dataCardTagTag = [];
+const dataCardEventInfos = [];
 
-  return data;
-};
+prepareMerge(dataCard.card.infos, dataCardInfos, 'label');
```

Comment:
It took me a while to track back all the reasoning that led you to this implementation, it feels to me that we are overdoing here, basically all of this is done for then merging the specs with keys and values that we almost entirely set in the story file.
So that all this processing would just not be needed in case we would simply pass the object to the story without merging it with the specs.

Tell me if this makes sense to you or if it's totally unclear, anyway in the code you currently have duplicated operations apparently.
What you do in the forEach would be normally done using map in javascript: https://www.w3schools.com/jsref/jsref_map.asp 
and the parameters name you used are very confusing cause you actually have two arrays but you name them "..ArrayName"

In my opinion we could reduce the amount of code and processing of these "small" amount of data and we should not set in the story "values" that we receive from the specs, but the knobs are working fine and i know you already worked a lot on this, so i'll let you decide whether to just "fix" what really needs to be fixed or to look for a simpler way to achieve the same result. ;)